### ISSUE TITLE
Crash when closing desktop notification on Windows

### ISSUE THREAD

**Issue by dbkr** [electron/electron#15251]
* Output of `node_modules/.bin/electron --version`: 3.0.4
* Operating System (Platform and Version): Windows 7, 32 & 64 bit
* Output of `node_modules/.bin/electron --version` on last known working Electron version (if applicable): 2.0.8

**Expected Behavior**
Calling .close() on a Notification object should cause the notification to no longer be shown to the user, or if it isn't shown, do nothing.

**Actual behavior**
Calling .close() on a Notification object after having displayed a notification will cause the electron process to crash with a null pointer exception.

**To Reproduce**
Clone and run https://github.com/dbkr/electron-quick-start then click the button labelled 'crash'.

Basically, show 2 notifications with the same 'tag' using `new global.Notification()`. Observe that the `onclose` handler of the first notification is called when the second is displayed. When calling `.close()` on the *second* notification (ie. the one whose `onclose` has *not* been called), electron will crash with a NPE in desktop_notification_controller.cc:374 -  `if (data_->controller)` in `DesktopNotificationController::Notification::Close()`.

**Screenshots**
None

**Additional Information**
After electron crashes, it also seems to leave an electron.exe in the background using about 20% of a core.

**Comment by dbkr** [comment]
I've added a minimal repo that demonstrates this. It turns out it only happens when displaying multiple notifications with the same 'tag'. I've updated the description appropriately.

This isn't reproducible with 4.0.0-beta.3, and it turns out we were using `tag` erroneously anyway, so this is no longer an issue for us.

**Comment by sofianguy** [comment]
If it is fixed in version 4.x, please upgrade. We are no longer supporting versions earlier than v4.2.x. Let us know if this is still an issue for you and we'd be happy to reopen this issue.
